server/src/main/java/org/solarbank/server/dto/CalculateRequest.java

```diff
@@ -0,0 +1,27 @@
+package org.solarbank.server.dto;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.DecimalMax;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import lombok.Data;
+
+@Data
+@JsonNaming(PropertyNamingStrategies.UpperCamelCaseStrategy.class)
+public class CalculateRequest {
```

Comment:
I think we need to add unit tests for all of your DTOs which apply validation annotations to check that they are working and to assert that the messages that are generated by the validation errors are what you expect. 

Here is an example applied to the `Location` dto:

```
import jakarta.validation.ValidatorFactory;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.solarbank.server.dto.Location;

import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class LocationTest {
    private Validator validator;

    @BeforeEach
    public void setUp() {
        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
        validator = factory.getValidator();
    }

    @Test
    public void testValidLocation() {
        Location location = new Location();
        location.setLongitude(100.0);
        location.setLat(50.0);

        Set<ConstraintViolation<Location>> violations = validator.validate(location);
        assertTrue(violations.isEmpty());
    }

    @Test
    public void testNullLongitude() {
        Location location = new Location();
        location.setLongitude(null);
        location.setLat(50.0);

        Set<ConstraintViolation<Location>> violations = validator.validate(location);
        assertEquals(1, violations.size());
        assertEquals("longitude must be provided", violations.iterator().next().getMessage());
    }
}
```

Its also worth noting that we should also move all of our custom validation message strings to an enum so that we are not typing them out in the tests.

Reply:
added enum for all the custom validation message strings. unit tests added for all dto with validation